File: sdks/android/src/main/kotlin/dev/nucleus/models/NucleusOrganization.kt
package dev.nucleus.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents a Nucleus organization the user belongs to.
 */
@Serializable
data class NucleusOrganization(
    val id: String,
    val name: String,
    val slug: String? = null,
    @SerialName("image_url")
    val imageUrl: String? = null,
    @SerialName("created_at")
    val createdAt: Long? = null,
    @SerialName("updated_at")
    val updatedAt: Long? = null,
    val role: String? = null,
    @SerialName("members_count")
    val membersCount: Int? = null,
)

File: sdks/swift/Sources/NucleusSwift/Models/NucleusOrganization.swift
import Foundation

/// Represents an organization the user belongs to.
public struct NucleusOrganization: Codable, Identifiable, Equatable, Sendable {
    public let id: String
    public let name: String
    public let slug: String?
    public let logoURL: String?
    public let createdAt: Date?
    public let updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case slug
        case logoURL = "logo_url"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}
